src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -93,6 +93,12 @@ public bool TryGetConfig([NotNullWhen(true)] out RuntimeConfig? runtimeConfig)
             if (ConfigLoader.TryLoadKnownConfig(out RuntimeConfig? config, replaceEnvVar: true))
             {
                 _runtimeConfig = config;
+
+                // Set up the managed identity dictionary with key values for each datasource for the access token. By default, the value is null.
+                foreach ((string dataSourceName, _) in _runtimeConfig.GetDataSourceNamesToDataSourcesIterator())
```

Comment:
Setting up of ManagedIdentityAccessToken to null shouldn't be inside this function. We call this TryGetConfig function multiple times, if this gets called after ManagedIdentityAccessToken had been set, it will override the value to null. 

Please move this code out to a separate function InitializeManagedIdentityAcessToken, and call that only once when u need to. or do it in constructor.

Reply:
This function does a check on _runtimeConfig to check if it is null. only when its not null, it tries to load a new config from the loader. When a new config is loaded, it could be loaded with different number of datasources or there could be changes. shouldnt it then correctly reset the ManagedIdentityAccessToken?Maybe could change this to a update if exists else add but i think the logic should still be here. what are your thoughts?

Reply:
I just rethought this, I guess my logic is wrong regardless because now we are only setting this dictionary up when a tryGet call is actually being made, which is not the right thing here. I will move this to the initializewithAccessToken.